aws-events-rule/src/main/java/software/amazon/events/rule/UpdateHandler.java

```diff
@@ -0,0 +1,105 @@
+package software.amazon.events.rule;
+
+import org.apache.commons.collections4.CollectionUtils;
+import software.amazon.awssdk.services.cloudwatchevents.CloudWatchEventsClient;
+import software.amazon.awssdk.services.cloudwatchevents.model.Target;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+import java.util.ArrayList;
+
+public class UpdateHandler extends BaseHandlerStd {
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<CloudWatchEventsClient> proxyClient,
+        final Logger logger) {
+
+        this.logger = logger;
+
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+
+            // STEP 1 [check if resource already exists]
+            .then(progress ->
+                proxy.initiate("AWS-Events-Rule::Update::PreUpdateCheck", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                    .translateToServiceRequest(Translator::translateToDescribeRuleRequest)
+                    .makeServiceCall((awsRequest, client) -> describeRule(awsRequest, client, logger, request.getStackId()))
+                    .handleError(this::handleError)
+                    .progress()
+            )
+
+            // STEP 2 [update the rule]
+            .then(progress ->
```

Comment:
we only update the rule if a property has changed. There should be a conditional statement checking for updated properties before entering the callgraph